File: my-folder/1392-find-the-difference-of-two-arrays/solution.py
import numpy as np
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        result = []
        result.append(list(set(nums1[i] for i in range(len(nums1)) if not any(nums1[i] == nums2[j] for j in range(len(nums2))))))
        result.append(list(set(nums2[i] for i in range(len(nums2)) if not any(nums2[i] == nums1[j] for j in range(len(nums1))))))

        return result


        

File: my-folder/1392-find-the-difference-of-two-arrays/solution.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        vector<vector<int>>res;

        unordered_set<int>s1(nums1.begin(), nums1.end());
        unordered_set<int>s2(nums2.begin(), nums2.end());

        vector<int>d1;
        vector<int>d2;

        for (int num: s2) {
            if (s1.find(num) == s1.end()) {
                d1.push_back(num);
            }
        }

        for (int num: s1) {
            if (s2.find(num) == s2.end()) {
                d2.push_back(num);
            }
        }

        res = {d2, d1};

        return res;
    }
};
